crypto/src/mls/conversation/conversation_guard.rs

```diff
@@ -0,0 +1,63 @@
+// this isssue is about creating this helper struct,
+// but we'll see the dead code warning until we actually use it in follow-up work
+#![expect(dead_code)]
+
+use async_lock::{RwLockReadGuard, RwLockWriteGuard};
+use mls_crypto_provider::MlsCryptoProvider;
+
+use crate::{context::CentralContext, group_store::GroupStoreValue, RecursiveError};
+
+use super::{commit::MlsCommitBundle, Error, MlsConversation, Result};
+
+/// A Conversation Guard wraps a `GroupStoreValue<MlsConversation>`.
+///
+/// By doing so, it permits mutable accesses to the conversation. This in turn
+/// means that we don't have to duplicate the entire `MlsConversation` API
+/// on `CentralContext`.
+pub(crate) struct ConversationGuard {
+    inner: GroupStoreValue<MlsConversation>,
+    central_context: CentralContext,
+}
+
+impl ConversationGuard {
+    pub(crate) fn new(inner: GroupStoreValue<MlsConversation>, central_context: CentralContext) -> Self {
```

Comment:
Why do we take ownership of the central context?

Reply:
They're cheap to clone (wrapping an `Arc` internally), and retaining ownership here means that many function signatures get simpler down the line.